src/util/triplea/mapXmlCreator/MapPropertiesPanel.java

```diff
@@ -0,0 +1,278 @@
+    stepActionPanel.add(labelMapName, gridBadConstLabelMapName);
+
+    textFieldMapName.addFocusListener(new FocusListener() {
+
+      @Override
+      public void focusLost(FocusEvent e) {
+        MapXmlHelper.putXmlStrings("info_@name", textFieldMapName.getText());
+      }
+
+      @Override
+      public void focusGained(FocusEvent e) {}
+    });
+    textFieldMapName.setMaximumSize(new Dimension(0, 0));
+    textFieldMapName.setColumns(30);
+    GridBagConstraints gridBadConstTextFieldMapName = (GridBagConstraints) gridBadConstLabelMapName.clone();
+    gridBadConstTextFieldMapName.anchor = GridBagConstraints.NORTH;
+    gridBadConstTextFieldMapName.gridx = 1;
+    textFieldMapName.setMaximumSize(textFieldMapName.getPreferredSize());
+    stepActionPanel.add(textFieldMapName, gridBadConstTextFieldMapName);
+
+    GridBagConstraints gridBadConstLabelMapNameExample = (GridBagConstraints) gridBadConstLabelMapName.clone();
+    gridBadConstLabelMapNameExample.anchor = GridBagConstraints.NORTHWEST;
+    gridBadConstLabelMapNameExample.gridx = 2;
+    stepActionPanel.add(labelMapNameExample, gridBadConstLabelMapNameExample);
+
+    // Map Version
+    stepActionPanel.add(labelMapVersion, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 2, 2));
+
+    textFieldMapVersion.addFocusListener(new FocusListener() {
+
+      @Override
+      public void focusLost(FocusEvent e) {
+        MapXmlHelper.putXmlStrings("info_@version", textFieldMapVersion.getText());
+      }
+
+      @Override
+      public void focusGained(FocusEvent e) {}
+    });
+    textFieldMapVersion.setMaximumSize(new Dimension(0, 0));
+    textFieldMapVersion.setColumns(30);
+    stepActionPanel.add(textFieldMapVersion, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 1, 2));
+
+
+    stepActionPanel.add(labelMapVersionExample, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapNameExample, 2, 2));
+
+    // Resource Name
+    stepActionPanel.add(labelResourceName, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 0, 3));
+
+    textFieldResourceName.addFocusListener(new FocusListener() {
+
+      @Override
+      public void focusLost(FocusEvent e) {
+        MapXmlHelper.addResourceList(0, textFieldResourceName.getText());
+      }
+
+      @Override
+      public void focusGained(FocusEvent e) {}
+    });
+    textFieldResourceName.setMaximumSize(new Dimension(0, 0));
+    textFieldResourceName.setColumns(30);
+    stepActionPanel.add(textFieldResourceName, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 1, 3));
+
+    stepActionPanel.add(labelResourceNameExample,
+        MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapNameExample, 2, 3));
+
+    // Map Image File
+    stepActionPanel.add(labelMapImageFile, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 0, 4));
+
+    textFieldMapImageFile.setEnabled(false);
+    textFieldMapImageFile.setMaximumSize(new Dimension(0, 0));
+    textFieldMapImageFile.setColumns(30);
```

Comment:
I extracted a local variable, but I do not know why it has to be 30. I
had quite some trouble with the UI as I am no expert in this field.

Am 11.04.2016 um 00:09 schrieb Dan Van Atta:
>
> In src/util/triplea/mapXmlCreator/MapPropertiesPanel.java
> <https://github.com/triplea-game/triplea/pull/598#discussion_r59142540>:
>
> > +      @Override
> > +      public void focusGained(FocusEvent e) {}
> > +    });
> > +    textFieldResourceName.setMaximumSize(new Dimension(0, 0));
> > +    textFieldResourceName.setColumns(30);
> > +    stepActionPanel.add(textFieldResourceName, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 1, 3));
> > +
> > +    stepActionPanel.add(labelResourceNameExample,
> > +        MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapNameExample, 2, 3));
> > +
> > +    // Map Image File
> > +    stepActionPanel.add(labelMapImageFile, MapXmlUIHelper.getGBCCloneWith(gridBadConstLabelMapName, 0, 4));
> > +
> > +    textFieldMapImageFile.setEnabled(false);
> > +    textFieldMapImageFile.setMaximumSize(new Dimension(0, 0));
> > +    textFieldMapImageFile.setColumns(30);
>
> magic number '30', please extract to a named constant and consolidate
> the cases that refer to the same value.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/triplea-game/triplea/pull/598/files/1420ca8d2fed62a1fa0ecccd1c212a5efef61778..8eddebfb8bb8b20baf4e726edae241ea5aeedd3b#r59142540>
>